src/platform/Ameba/ConfigurationManagerImpl.cpp

```diff
@@ -36,9 +36,25 @@ namespace DeviceLayer {
 
 using namespace ::chip::DeviceLayer::Internal;
 
-/** Singleton instance of the ConfigurationManager implementation object.
+/** Singleton pointer to the ConfigurationManager implementation.
  */
-ConfigurationManagerImpl ConfigurationManagerImpl::sInstance;
+ConfigurationManager * gInstance;
+
+ConfigurationManager & ConfigurationMgr()
+{
+    return *gInstance;
+}
+
+void SetConfigurationMgr(ConfigurationManager & configurationManager)
```

Comment:
how about passing in a pointer instead? passing a reference and then getting the reference address seems odd.

Reply:
I've made that change, and specified that passing in nullptr will do nothing.